plugins/modules/azure_rm_storageaccount.py

```diff
@@ -1715,12 +1841,50 @@ def update_static_website(self):
 
     def set_network_acls(self):
         try:
-            parameters = self.storage_models.StorageAccountUpdateParameters(network_rule_set=self.network_acls)
-            self.storage_client.storage_accounts.update(self.resource_group,
-                                                        self.name,
-                                                        parameters)
+            models = self.storage_models
+            nacls = self.network_acls or {}
+
+            def build_network_rule_set(nacls_local):
+                vnet_rules = None
+                if nacls_local.get('virtual_network_rules'):
+                    vnet_rules = [
+                        models.VirtualNetworkRule(
+                            virtual_network_resource_id=r.get('id'),
+                            action=r.get('action', 'Allow')
+                        ) for r in (nacls_local.get('virtual_network_rules') or [])
+                    ]
+
+                ip_rules = None
+                if nacls_local.get('ip_rules'):
+                    ip_rules = [
+                        models.IPRule(
+                            ip_address_or_range=r.get('value'),
+                            action=r.get('action', 'Allow')
+                        ) for r in (nacls_local.get('ip_rules') or [])
+                    ]
+
+                resource_access_rules = None
+                if nacls_local.get('resource_access_rules'):
+                    resource_access_rules = [
+                        models.ResourceAccessRule(
+                            resource_id=r.get('resource_id'),
+                            tenant_id=r.get('tenant_id')
+                        ) for r in (nacls_local.get('resource_access_rules') or [])
+                    ]
```

Comment:
Same here, why do we need this explicit handling? If the previous code works fine, it means at least for the `ip_rules` and vnet rules, they don't need the explict handling above?